pen/views.py

```diff
@@ -0,0 +1,75 @@
+from django.contrib.auth.forms import UserCreationForm
+from django.shortcuts import render, redirect
+from .models import Memo
+from django.contrib.auth import authenticate, logout
+from django.contrib.auth import login
+from django.views.decorators.http import require_http_methods
+
+
+def memo(request):
```

Comment:
Pytanie, czy chcemy udostępnić ten widok niezalogowanemu użytkownikowi. Nawet jeżeli założymy, że użytkownik niezalogowany może widzieć planszę, to nie powinien mieć chyba udostępnionego przycisku zapisz w szablonie. No bo jak taki niezalogowany uzytkownik zrobi notatkę i ją zapisze, a potem wejdzie na stronę inny niezalogowany użytkownik. Wyświetli mu się wtedy notatka poprzedniego użytkownika, którą z łatwością będzie mógł nadpisać, jeżeli sam zapiszę notatkę (bo niezalogowany użytkownik to po prostu użytkownik anonimowy i jak każdy inny użytkownik może mieć tylko jedną zapisaną notatkę). Więc o ile takie rozwiązanie nie jest zamierzone to najlepiej będzie ograniczyć dostęp do tego widoku tylko dla użytkowników zalogowanych - służy do tego dekorator login_required. (można ewentualnie rozważyć w szablonie memo.html odsłanianie przycisku "zapisz" tylko dla użytkownika zalogowanego, ale to determinuje też zmiany w js, bo w takim przypadku użytkownik niezalogowany będzie miał 37 przycisków, podczas gdy użytkownik zalogowany będzie miał 38 przycisków (różnica o przycisk zapisz)).

Reply:
Widok nie jest udostępniany niezalogowanym użytkownikom. Ograniczono dostęp do tego widoku tylko dla użytkowników zalogowanych - posłużono się tutaj dekoratorem login_required.